build.xml

```diff
@@ -22,18 +23,339 @@
 		<pathelement location="${env.CATALINA_HOME}/lib/catalina-ant.jar"/>
 	</path>
 
-	<target name="clean_and_tomcat_deploy" description="Deploy prepared WAR file from project dir to Tomcat webapps folder">
-		<delete file="${env.CATALINA_HOME}/webapps/ScadaBR.war"></delete>
-		<delete dir="${env.CATALINA_HOME}/webapps/ScadaBR"></delete>
-		<copy file="ScadaBR.war" todir="${env.CATALINA_HOME}/webapps" overwrite="true">
+	<target name="init-core" description="Prepare project before build">
+		<delete dir="${build-project.dir}"/>
+		<mkdir dir="${build-project.dir}"/>
+		<mkdir dir="${build-project.src.dir}"/>
+		<mkdir dir="${build-project.test.dir}"/>
+		<mkdir dir="${build-project.log.dir}"/>
+		<copy includeemptydirs="false" todir="${build-project.src.dir}">
+			<fileset dir="src">
+				<exclude name="**/*.launch"/>
+				<exclude name="**/*.xtend"/>
+				<exclude name="**/*.java"/>
+			</fileset>
+		</copy>
+		<copy includeemptydirs="false" todir="${build-project.test.dir}">
+			<fileset dir="test">
+				<exclude name="**/*.launch"/>
+				<exclude name="**/*.xtend"/>
+				<exclude name="**/*.java"/>
+			</fileset>
+		</copy>
+	</target>
+
+	<target name="build-core" description="Build Scada-LTS core application">
+		<record action="start" name="${build-project.log.src.name}"/>
+		<echo message="${ant.project.name}: ${ant.file}"/>
+		<javac debug="true" debuglevel="${debuglevel}" destdir="${build-project.src.dir}" includeantruntime="false" source="${source}" target="${target}">
+			<src path="src"/>
+			<classpath refid="scada-lts.classpath"/>
+		</javac>
+		<record action="stop" name="${build-project.log.src.name}"/>
+		<record action="start" name="${build-project.log.test.name}"/>
+		<javac debug="true" debuglevel="${debuglevel}" destdir="${build-project.test.dir}" includeantruntime="false" source="${source}" target="${target}">
+			<src path="test"/>
+			<classpath refid="scada-lts.classpath"/>
+			<classpath refid="test-unit.classpath"/>
+			<classpath refid="benchmark.classpath"/>
+		</javac>
+		<record action="stop" name="${build-project.log.test.name}"/>
+	</target>
+
+	<target name="init-static-dependencies" description="Install JavaScript dependencies for Scada-LTS using nodejs.">
+		<exec dir="./WebContent/resources" executable="npm.cmd" osfamily="windows">
+			<arg value="install"/>
+		</exec>
+		<exec dir="./WebContent/resources" executable="npm" osfamily="unix">
+			<arg value="install"/>
+		</exec>
+	</target>
+
+	<target name="init-ui-dependencies" description="Install frontend application dependencies">
+		<exec dir="./scadalts-ui" executable="npm.cmd" osfamily="windows">
+			<arg value="install"/>
+		</exec>
+		<exec dir="./scadalts-ui" executable="npm" osfamily="unix">
+			<arg value="install"/>
+		</exec>
+	</target>
+
+	<target name="build-ui" description="Build frontend application">
```

Comment:
Perhaps it would be nice to create logs for the main targets? (`<record/>`)

Reply:
It's good idea :)